gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/LoadConfiguration.java

```diff
@@ -0,0 +1,381 @@
+      this.ignoreUnknownValues = ignoreUnknownValues;
+      return this;
+    }
+
+    /**
+     * Sets which entity properties to load into BigQuery from a Cloud Datastore backup. This field
+     * is only used if the source format is set to {@code DATASTORE_BACKUP}. Property names are case
+     * sensitive and must be top-level properties. If no properties are specified, BigQuery loads
+     * all properties. If any named property isn't found in the Cloud Datastore backup, an invalid
+     * error is returned in the job result.
+     */
+    public Builder projectionFields(List<String> projectionFields) {
+      this.projectionFields =
+          projectionFields != null ? ImmutableList.copyOf(projectionFields) : null;
+      return this;
+    }
+
+    public LoadConfiguration build() {
+      return new LoadConfiguration(this);
+    }
+  }
+
+  private LoadConfiguration(Builder builder) {
+    this.destinationTable = checkNotNull(builder.destinationTable);
+    this.createDisposition = builder.createDisposition;
+    this.writeDisposition = builder.writeDisposition;
+    this.formatOptions = builder.formatOptions;
+    this.maxBadRecords = builder.maxBadRecords;
+    this.schema = builder.schema;
+    this.ignoreUnknownValues = builder.ignoreUnknownValues;
+    this.projectionFields = builder.projectionFields;
+  }
+
+  /**
+   * Returns the destination table to load the data into.
+   */
+  public TableId destinationTable() {
+    return destinationTable;
+  }
+
+  /**
+   * Returns whether the job is allowed to create new tables.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.load">
+   *     Jobs: Load Configuration</a>
+   */
+  public CreateDisposition createDisposition() {
+    return this.createDisposition;
+  }
+
+  /**
+   * Returns the action that should occur if the destination table already exists.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.load">
+   *     Jobs: Load Configuration</a>
+   */
+  public WriteDisposition writeDisposition() {
+    return writeDisposition;
+  }
+
+  /**
+   * Returns additional properties used to parse CSV data (used when {@link #format()} is set
+   * to CSV). Returns {@code null} if not set.
+   */
+  public CsvOptions csvOptions() {
+    return formatOptions instanceof CsvOptions ? (CsvOptions) formatOptions : null;
+  }
+
+  /**
+   * Returns the maximum number of bad records that BigQuery can ignore when running the job. If the
+   * number of bad records exceeds this value, an invalid error is returned in the job result.
+   * By default no bad record is ignored.
+   */
+  public Integer maxBadRecords() {
+    return maxBadRecords;
+  }
+
+  /**
+   * Returns the schema for the destination table, if set. Returns {@code null} otherwise.
+   */
+  public Schema schema() {
+    return schema;
+  }
+
+  /**
+   * Returns the format of the data files.
+   */
+  public String format() {
```

Comment:
Should this be `formatOptions`?  Also I like the javadoc in the builder for this functionality, can we copy it over for this method?